File: src/main/java/xyz/oribuin/eternalclaims/hook/economy/EconomyPlugin.java
package xyz.oribuin.eternalclaims.hook.economy;

import org.jetbrains.annotations.NotNull;

import java.util.UUID;

public interface EconomyPlugin {

    /**
     * @param player The player
     * @return The balance of the player
     */
    double getBalance(@NotNull UUID player);

    /**
     * @param player The player
     * @param amount The amount to withdraw
     */
    void withdraw(@NotNull UUID player, double amount);

    /**
     * @param player The player
     * @param amount The amount to deposit
     */
    void deposit(@NotNull UUID player, double amount);

    /**
     * @param player The player
     * @param amount The amount to check
     * @return If the player has the amount
     */
    boolean has(@NotNull UUID player, double amount);

}

File: src/main/kotlin/dev/oribuin/eternalclaims/hook/economy/EconomyPlugin.kt
package dev.oribuin.eternalclaims.hook.economy

import org.bukkit.OfflinePlayer

interface EconomyPlugin {
    /**
     * Check if the economy plugin is enabled
     *
     * @return If the economy plugin is enabled
     */
    fun enabled(): Boolean

    /**
     * Take money from a player
     *
     * @param player The player
     * @param amount The amount
     * @return If the transaction was successful
     */
    fun take(player: OfflinePlayer, amount: Double): Boolean

    /**
     * Give a player money
     *
     * @param player The player
     * @param amount The amount
     * @return If the transaction was successful
     */
    fun give(player: OfflinePlayer, amount: Double): Boolean

    /**
     * Get the balance of a player
     *
     * @param player The player
     * @return The balance
     */
    fun balance(player: OfflinePlayer): Double

    /**
     * Check if a player has a certain amount of money
     *
     * @param player The player
     * @param amount The amount
     * @return If the player has the amount
     */
    fun has(player: OfflinePlayer, amount: Double): Boolean
}